android/filament-android/src/main/cpp/MaterialInstance.cpp

```diff
@@ -131,9 +155,26 @@ Java_com_google_android_filament_MaterialInstance_nSetBooleanParameterArray(JNIE
     MaterialInstance* instance = (MaterialInstance*) nativeMaterialInstance;
 
     const char* name = env->GetStringUTFChars(name_, 0);
-    size_t size = (size_t) element + 1;
     jboolean* v = env->GetBooleanArrayElements(v_, NULL);
-    instance->setParameter(name, (bool*) (v + offset * size), (size_t) (count * size));
+
+    // NOTE: In C++, bool has an implementation-defined size. Here we assume
+    // it has the same size as jboolean, which is 1 byte.
+
+    switch ((BooleanElement) element) {
```

Comment:
I'd rather we keep the previous code with a comment, I don't see the benefit of doing it this way. It generates more code too and I don't think it's particularly more readable.

Reply:
We actually need switch statements in order to call the correct overloads. The previous code always called the scalar overloads for bool/int/float, which was incorrect for vector types.

If you'd like, I can replace the enums with hardcoded constants. Note that these enums not only improve readability, they also improve consistency with the corollary Java code.

Reply:
Ah right, that array of 1 problem again. Ok sounds good.